api/src/main/java/com/epam/pipeline/manager/datastorage/providers/nfs/NFSQuotasMonitor.java

```diff
@@ -0,0 +1,209 @@
+import com.epam.pipeline.config.JsonMapper;
+import com.epam.pipeline.entity.BaseEntity;
+import com.epam.pipeline.entity.datastorage.DataStorageType;
+import com.epam.pipeline.entity.datastorage.FileShareMount;
+import com.epam.pipeline.entity.datastorage.LustreFS;
+import com.epam.pipeline.entity.datastorage.MountType;
+import com.epam.pipeline.entity.datastorage.NFSStorageMountStatus;
+import com.epam.pipeline.entity.datastorage.StorageUsage;
+import com.epam.pipeline.entity.datastorage.nfs.NFSQuotaNotificationEntry;
+import com.epam.pipeline.entity.datastorage.nfs.NFSDataStorage;
+import com.epam.pipeline.entity.datastorage.nfs.NFSQuota;
+import com.epam.pipeline.entity.metadata.MetadataEntry;
+import com.epam.pipeline.entity.metadata.PipeConfValue;
+import com.epam.pipeline.entity.security.acl.AclClass;
+import com.epam.pipeline.manager.datastorage.DataStorageManager;
+import com.epam.pipeline.manager.datastorage.FileShareMountManager;
+import com.epam.pipeline.manager.datastorage.lustre.LustreFSManager;
+import com.epam.pipeline.manager.metadata.MetadataManager;
+import com.epam.pipeline.manager.search.SearchManager;
+import com.fasterxml.jackson.core.type.TypeReference;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections4.CollectionUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+@Slf4j
+public class NFSQuotasMonitor {
+
+    private static final int GB_TO_BYTES = 1024 * 1024 * 1024;
+    private static final String SIZE_QUOTA_GB = "GB";
+    private static final String SIZE_QUOTA_PERCENTS = "PERCENT";
+    private static final int PERCENTS_MULTIPLIER = 100;
+
+    private final DataStorageManager dataStorageManager;
+    private final SearchManager searchManager;
+    private final MetadataManager metadataManager;
+    private final FileShareMountManager fileShareMountManager;
+    private final LustreFSManager lustreManager;
+    private final MessageHelper messageHelper;
+    private final String emailNotificationAction;
+    private final String readOnlyAction;
+    private final String disableMountAction;
+    private final String notificationsKey;
+
+    public NFSQuotasMonitor(final DataStorageManager dataStorageManager,
+                            final SearchManager searchManager,
+                            final MetadataManager metadataManager,
+                            final FileShareMountManager fileShareMountManager,
+                            final LustreFSManager lustreManager,
+                            final MessageHelper messageHelper,
+                            final @Value("${data.storage.nfs.quota.action.email:EMAIL}")
```

Comment:
Can't we use enum for actions?

Reply:
Done, new `StorageQuotaAction` enum is introduced